diff --git a/src/main/java/com/peng/controller/CommentControllers.java b/src/main/java/com/peng/controller/CommentControllers.java
index 456f5c2..a998f98 100644
--- a/src/main/java/com/peng/controller/CommentControllers.java
+++ b/src/main/java/com/peng/controller/CommentControllers.java
@@ -8,6 +8,7 @@ import com.peng.entity.Comment;
 import com.peng.service.IBlogService;
 import com.peng.service.ICommentService;
 import com.peng.util.IpUtil;
+import com.peng.util.RedisUtil;
 import com.rabbitmq.tools.json.JSONUtil;
 import org.springframework.amqp.rabbit.core.RabbitTemplate;
 import org.springframework.beans.factory.annotation.Autowired;
@@ -32,6 +33,9 @@ public class CommentControllers {
     @Autowired
     private RabbitTemplate rabbitTemplate;
 
+    @Autowired
+    private RedisUtil redisUtil;
+
     @MyLog
     @GetMapping("/comments/{blId}")
     public String comments(@PathVariable Long blId, Model model) {
@@ -46,13 +50,14 @@ public class CommentControllers {
     @MyLog
     @PostMapping("/comments")
     public String postComments(Comment comment, HttpServletRequest request) {
-        //保存评论
         if (comment.getParentId() <= -1)
             comment.setParentId(null);
         String ipAddress = IpUtil.getIpAddress(request);
         comment.setIpAddress(ipAddress);
         iCommentService.saveOrUpdate(comment);
         rabbitTemplate.convertAndSend("msg-event-exchange", "msg.wx-pn", formatWxMsg(comment));
+        redisUtil.del("com.peng.service.Impl.CacheServiceImpl.getCommentNum");
+        redisUtil.deleteByPattern("com.peng.service.Impl.CacheServiceImpl.getIndexPage-*");
         return "redirect:/comments/" + comment.getBlId();
     }
 
diff --git a/src/main/java/com/peng/util/RedisUtil.java b/src/main/java/com/peng/util/RedisUtil.java
index 341244e..d2a5f6d 100644
--- a/src/main/java/com/peng/util/RedisUtil.java
+++ b/src/main/java/com/peng/util/RedisUtil.java
@@ -76,6 +76,13 @@ public final class RedisUtil {
             }
         }
     }
+
+    public void deleteByPattern(String pattern) {
+        Set<String> keys = redisTemplate.keys(pattern);
+        if (keys != null && !keys.isEmpty()) {
+            redisTemplate.delete(keys);
+        }
+    }
     // ============================String=============================
 
     /**
